### DISCUSSION TITLE
Lights turn off briefly while changing colors, but not always

### DISCUSSION THREAD

**Discussion by ulisesftw** [liquidctl/liquidctl#358] [Q&A]
Hi! First of all thankyou for the great work and sorry if this isn't the right place to ask.

Im using **liquidctl** command with **subprocess.Popen()** in python (because I'm still new and its all I know) to control the color lights of a **Corsair Lighting Node Core**. I noticed that when changeing colors with the command 

**liquidctl set led color fixed hsv(120,100,100) --start-led 1 --maximum-leds 24**

sometimes the color changes instantly but most of the time goes black briefly. 
Seeing that it doesn't happens all the time I was wondering if there was a way to prevent this from happening. If you have any clue or another way to changing colors I'd appreciate it.

My end goal is to make my lights predict the weather just like those [blue and pink souvenir](https://www.google.com/search?q=souvenir+predice+clima&tbm=isch&ved=2ahUKEwir657t4d3xAhW8M7kGHY0NDaAQ2-cCegQIABAA&oq=souvenir+&gs_lcp=CgNpbWcQARgAMgQIIxAnMgIIADICCAAyAggAMgIIADICCAAyAggAMgIIADICCAAyAggAOgQIABBDOgYIABAKEBg6BggAEAUQHjoECAAQHjoGCAAQCBAeUORHWNZjYOl0aAZwAHgAgAE_iAHHBpIBAjE1mAEAoAEBqgELZ3dzLXdpei1pbWfAAQE&sclient=img&ei=lVTsYOvMJLzn5OUPjZu0gAo&bih=898&biw=1053&hl=en) (google search)

Thanks in advance to everyone.

**Comment by jonasmalacofilho** [comment]
I'm not sure.

@MarshallAsch, do you know what's causing this (I'm guessing it's related to the reuploading of all saved effects), if there's something we can do about it, or if there are any workarounds (maybe clearing all effects before the new update)?

---

Another thing that comes to mind is that, since you're using Python, you could use the liquidctl APIs directly, instead of invoking the CLI with `subprocess.Popen()`.

**Reply by MarshallAsch** [reply]
I am pretty sure it is related to the re upload part, since the set colour for the commander pro has a limited number of effects that can be applied at a time. ie it will send all the messages to set the colours even of a later one over rides an earlier one. 

That can't really be fixed  for hardware mode because of of the commands that gets sent is a `clear` command followed by a `set` command. It could be done using the software control mode, which we did not implement because it requires sending messages every ~20 seconds. Since liquidctl does not run as a daemon it was not implemented.

**Reply by ulisesftw** [reply]
Thanks for the replies. I will eventually move on to use the python api directly when I understand it a little better.

Right now Im doing a **set led color clear** every 8 color changes which seems to be the limit but I found out that the blackout happens either I cleared or not. But since doesnt happens all the time, there are color changes that go seamless I was hoping for a way to force that. 
Again thanks!
